handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java

```diff
@@ -3245,6 +3250,71 @@ protected void engineInit(ManagerFactoryParameters managerFactoryParameters) {
         }
     }
 
+    @Test
+    public void testMasterKeyLogging() throws Exception {
+
+        /*
+         * At the moment master key logging is not supported for conscrypt
+         */
+        if (serverSslContextProvider() instanceof OpenSSLProvider) {
+            return;
+        }
+
```

Comment:
Also add the following code to ensure the test is skipped when JDK9+ is used and its the JDK implementation (as reflection is not allowed there by default):

```
if (sslServerProvider() == SslProvider.JDK && PlatformDependent.javaVersion() > 8) {
    return;
}
```

Reply:
Yea I'll do that -- I figured it out at the same time.. 
Had to install java and run `mvn.surefire.debug` and attach to the process.
IntelliJ is shitting the bed when I changed the SDK to 11....

I was going do `Assume.assumeTrue(TlsMasterKeyHandler.isSunSslEngineAvailable());` but I like the explicit platform version check.

Reply:
Actually I think your idea is even better...

So just make it :

```java
if (sslServerProvider()) {
    Assume.assumeTrue(TlsMasterKeyHandler.isSunSslEngineAvailable());
}
```

Reply:
I opted for something in the middle.
I actually like encoding that it's specific to the JDK version --
`Assume.assumeTrue(TlsMasterKeyHandler.isSunSslEngineAvailable())` might hide cases where it should be available on JDK8.